### PR TITLE
Upgrade go-boost-utils

### PR DESCRIPTION
**Pull request by avalonche**
## 📝 Summary

Upgrade relay code to the latest go-boost-utils as the library migrated to attestant types.

## 📚 References

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [ ] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

**Review by jtraglia — CHANGES_REQUESTED** [review]
Here's my initial review. Overall looks pretty good.

File: services/api/optimistic_test.go
Comment (avalonche) [reply services/api/optimistic_test.go:33]: `01234567890123456789012345678901` had an invalid length, `0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2` is one of the example values from the [beacon-API](https://github.com/ethereum/beacon-APIs/blob/20980bf18dc71306c9157a0b7f82c164bf84f76d/types/primitive.yaml#L70)

File: services/api/types_test.go
Comment (avalonche) [reply services/api/types_test.go:10]: I'll raise a new pr for the renaming as this cuts across more than changes to go-boost-utils

File: services/api/optimistic_test.go
Comment (jtraglia) [reply services/api/optimistic_test.go:33]: Gotcha. Thanks.

File: services/api/types_test.go
Comment (jtraglia) [reply services/api/types_test.go:10]: Okay. Sounds good. Works for me.

File: go.mod
Comment (jtraglia) [reply go.mod:1]: Ah I understand. Is the plan to use your fork until that PR has been merged & released? Or would you prefer we wait until it's upstreamed to merge this PR?

File: services/api/service.go
Comment (jtraglia) [reply services/api/service.go:939]: Thanks 👍

File: go.mod
Comment (avalonche) [reply go.mod:1]: This PR will be merged into the `develop` so it'll be more a WIP branch using my fork until the changes in `go-builder-client` stabilise. There are a few more changes needed in that library.

File: services/api/service_test.go
Comment (avalonche) [reply services/api/service_test.go]: everything should be cleaned up now

**Review by jtraglia — APPROVED** [review]
LGTM 👍 great work!

**Pull request merged**
